### PR TITLE
Confirm user

### PR DESCRIPTION
**Pull request by harissmansoor63**
https://app.asana.com/0/1190276726178297/1200157591260085/f

### CODE REVIEW

File: app/assets/javascripts/admin/products.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (hamzafida123) [inline app/assets/javascripts/admin/products.coffee]: You don't need to add files in your commits that you are not using. You can remove this file. Similar for "product.scss", "products_helper"

File: app/views/admin/users/edit.html.erb [hunk 2]
```diff
@@ -1,37 +1,50 @@
 <%= form_with(model: @user, url: admin_user_path(@user), method: 'patch' , local: true) do |f| %>
+  <% if @user.errors.any? %>
```
Comment (hamzafida123) [inline app/views/admin/users/edit.html.erb]: add a partial for error messages in shared folder and use that partial in all views.

**Review by hamzafida123 — APPROVED** [review]

File: app/controllers/admin/products_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+class Admin::ProductsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_product, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @products = Product.all
+  end
+
+  def new
+     @product = Product.new
```
Comment (faisalraza) [inline app/controllers/admin/products_controller.rb:10]: Indentation issue

File: app/controllers/admin/products_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  private
+
+  def product_params
+    params.require(:product).permit(:title, :price, :description, :status)
+  end
+
+  def set_product
+    @product = Product.find(params[:id])
+  end
+
```
Comment (faisalraza) [inline app/controllers/admin/products_controller.rb:48]: Extra line

File: app/views/admin/products/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<%= form_with(model: @product, url: admin_product_path(@product), method: 'patch' , local: true) do |f| %>
```
Comment (faisalraza) [inline app/views/admin/products/edit.html.erb:1]: Extra space after `'patch'`

File: app/views/admin/products/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    <%= f.select :status, ['Publish', 'Draft', 'Pending'] %>
+  </div>
+
+  <div class="field">
+    <div class= "clearfix"> <%= f.label :description %> </div>
+    <%= f.text_area :description, rows: "5", cols: "50", autofocus: true, autocomplete: "Description" %>
+  </div>
+
+  <div class="actions">
+      <%= f.submit "Update" %>
```
Comment (faisalraza) [inline app/views/admin/products/edit.html.erb:25]: Indentation issue

File: app/views/admin/products/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+  <div class="field">
+    <div class= "clearfix"> <%= f.label :status %> </div>
+    <%= f.select :status, ['Publish', 'Draft', 'Pending'] %>
+  </div>
+
+  <div class="field">
+    <div class= "clearfix"> <%= f.label :description %> </div>
+    <%= f.text_area :description, rows: "5", cols: "50", autofocus: true, autocomplete: "Description" %>
+  </div>
```
Comment (faisalraza) [inline app/views/admin/products/edit.html.erb:22]: Lets move the common fields in a partial ie, `admin/products/_form.html` and reuse it in new and edit views

File: app/views/admin/products/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,29 @@
+<%= form_with(model: @product, url: admin_products_path, method: 'post' , local: true) do |f| %>
```
Comment (faisalraza) [inline app/views/admin/products/new.html.erb:1]: space issue

File: app/views/admin/users/edit.html.erb [hunk 9]
```diff
@@ -1,37 +1,39 @@
...
-    <%= f.label :password_confirmation %><br />
+    <div class= "clearfix"> <%= f.label :password_confirmation %> </div>
     <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
   </div>
 
-  <p>
-    <%= f.submit %>
-  </p>
+ <div class="actions">
+      <%= f.submit "Update" %>
```
Comment (faisalraza) [inline app/views/admin/users/edit.html.erb:35]: Indentation issue

File: app/views/admin/users/edit.html.erb [hunk 10]
```diff
@@ -1,37 +1,39 @@
 <%= form_with(model: @user, url: admin_user_path(@user), method: 'patch' , local: true) do |f| %>
+  <%= render "shared/error_messages", resource: @user %>
+
   <div class="field">
-    <%= f.label :first_name %><br />
+    <div class= "clearfix"> <%= f.label :first_name %> </div>
```
Comment (faisalraza) [inline app/views/admin/users/edit.html.erb:5]: Extra space after `class=` in the whole file and in next few files
Lets remove space around label

File: app/views/devise/sessions/new.html.erb [hunk 11]
```diff
@@ -34,43 +34,44 @@
...
     </div>
 
     <div class="field">
-      <%= f.label :password_confirmation %><br />
+      <div class= "clearfix"> <%= f.label :password_confirmation %> </div>
       <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
     </div>
 
-    <p></p><br />Hint: The password should be at least eight characters long. <br /> To make it stronger, use upper and lower case letters, numbers, and symbols like ! " ? $ % ^ & ).</p>
+    <p> Hint: The password should be at least eight characters long. </p>
```
Comment (faisalraza) [inline app/views/devise/sessions/new.html.erb:66]: Lets remove extra spaces

File: app/views/layouts/_navbar.html.erb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+<nav class="navbar navbar-light bg-light">
+  <a class="navbar-brand" href="/">ProductMS</a>
+  <form class="form-inline">
+    <%= link_to 'Change Password',  set_password_path(current_user.id) %>
```
Comment (faisalraza) [inline app/views/layouts/_navbar.html.erb:4]: Extra space

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -1,5 +1,9 @@
 <head>
-	<title>ProductMS</title>
+  <% if current_user && current_user.admin? %>
```
Comment (faisalraza) [inline app/views/layouts/application.html.erb:2]: Do we need to add `current_user` check as we already authentication user in application controller level?

**Pull request merged**